### PR TITLE
Action_built: Add unittest for percentage lookup

### PR DESCRIPTION
**Pull request by zzam**
Testing the percentage lookup code fixed some days ago.

TODO: What else to test in here? and how?

### CODE REVIEW

File: src/action/action_built.cpp [hunk 1]
```diff
@@ -323,6 +323,10 @@ void COrder_Built::AiUnitKilled(CUnit &unit)
 
 static std::size_t FindCFramePercent(const std::vector<CConstructionFrame> &cframes, int percent)
 {
+	if (cframes.empty())
+	{
+		return -1;
```
Comment (Jarod42) [inline src/action/action_built.cpp:328]: returning `std::optional<std::size_t>` might be an option. that ensures the test for validity.

File: .editorconfig [hunk 2]
```diff
@@ -8,6 +8,7 @@ root = true
 charset = utf-8
 end_of_line = lf
 insert_final_newline = true
+trim_trailing_whitespace = true
```
Comment (Jarod42) [inline .editorconfig:11]: Better to split PR next time.

File: tests/stratagus/test_action_built.cpp [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+class tested_COrder_Built : public COrder_Built
+{
+public:
+	using COrder_Built::ProgressCounter;
+
+	int testUpdateConstrFrame_Percent(CUnit &unit, int percent, int initialFrame = 9999)
+	{
+		const int progressOnePercent = unit.Type->Stats[0].Costs[TimeCost] * 6;
+		return testUpdateConstrFrame_Progress(unit, percent * progressOnePercent, initialFrame);
+		}
```
Comment (Jarod42) [inline tests/stratagus/test_action_built.cpp:46]: Not formatted correctly.

File: tests/stratagus/test_action_built.cpp [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+		Frame = -5; // force re-calculation
+		ProgressCounter = progress;
+		UpdateConstructionFrame(unit);
+		return unit.Frame;
+	}
+};
+
+namespace {
+	void setFrames(std::vector<CConstructionFrame> &cframes,
+				   const std::vector<std::pair<int, int>>& vec)
```
Comment (Jarod42) [inline tests/stratagus/test_action_built.cpp:60]: Would `std::vector<CConstructionFrame> makeFrames(const std::vector<std::pair<int, int>>& vec)` make sense instead?

**Review by Jarod42 — APPROVED** [review]

File: .editorconfig (hunk 2 above)
Comment (zzam) [reply .editorconfig:11]: Created https://github.com/Wargus/stratagus/pull/648

File: tests/stratagus/test_action_built.cpp (hunk 3 above)
Comment (zzam) [reply tests/stratagus/test_action_built.cpp:46]: fixed

File: tests/stratagus/test_action_built.cpp (hunk 4 above)
Comment (zzam) [reply tests/stratagus/test_action_built.cpp]: changed. But does not look too different.

**Pull request merged**